posthog/settings.py

```diff
@@ -754,3 +754,5 @@ def show_toolbar(request):
     SERVICE_VERSION_REQUIREMENTS = SERVICE_VERSION_REQUIREMENTS + [
         ServiceVersionRequirement(service="clickhouse", supported_version=">=21.6.0,<21.7.0"),
     ]
+
+AUTO_START_SPECIAL_MIGRATIONS = get_from_env("AUTO_START_SPECIAL_MIGRATIONS", True, type_cast=str_to_bool)
```

Comment:
This is nice & I like it that we default to true, but note that often times when doing an update one will need to follow-up with another update that could mean the special migration gets interrupted. Which might not be that great, a 30min/1h delay could help here perhaps?

Additionally special migrations could break something and we might be pushing them out at a time that's inconvenient for the user. The later I've seen solved as providing upgrade windows for when upgrade operations can run (later we might do PostHog version upgrades magically too tapping into the same settings), e.g. DigitalOcean does that for k8s minor version patch upgrades optionally enabled. We don't need to do this now, but something to consider in the future/as a follow-up.

Reply:
actionable point: I'll at least add a bit of a delay